java/code/src/com/suse/manager/maintenance/MaintenanceManager.java

```diff
@@ -695,6 +696,71 @@ public boolean isActionInMaintenanceWindow(Action action, MaintenanceSchedule sc
         return false;
     }
 
+    private Collection<CalendarComponent> getCalendarForNow(MaintenanceSchedule ms) {
+        return getScheduleEventsAtDate(new Date(), ms, parseCalendar(ms.getCalendarOpt().get()));
+    }
+
+    /**
+     * Check if system is in maintenance mode
+     *
+     * @param server the server to check
+     * @return true when the action is inside of a maintenance window, otherwise falsegg
+     */
+    public static boolean checkIfInMaintenanceMode(MinionServer server) {
+        if (server.getMaintenanceScheduleOpt().isEmpty()) {
+            return true;
+        }
+        else {
+            MaintenanceManager mm = MaintenanceManager.instance();
+            Collection<CalendarComponent> cc = mm.getCalendarForNow(server.getMaintenanceScheduleOpt().get());
+            return !cc.isEmpty();
+        }
+    }
+
+    /**
+     * Log the number of servers skipped and if debugging is enabled list the server names
+     *
+     * @param servers the list of servers to log
+     */
+    public static void logSkippedMinions(List<MinionServer> servers) {
+       log.warn("Skipping action for " + servers.size() + " minions.");
+       if (log.isDebugEnabled()) {
+          String serverNames = servers.stream()
+          .map(m -> m.getName())
+          .collect(Collectors.joining(","));
+          log.debug("Skipped minion ids: " + serverNames);
+       }
+    }
+
+    /**
+     * Given a list of minions, sorts by maintenance mode status, logs skipped minions
+     *
+     * @param minions servers to check
+     * @return List of minions in maintenance mode
+     */
+    public static List<Long> systemIdsMaintenanceMode(List<MinionServer> minions) {
+        MaintenanceManager mm = MaintenanceManager.instance();
+        Set<MaintenanceSchedule> schedulesInMaintMode = minions.stream()
+            .flatMap(minion -> minion.getMaintenanceScheduleOpt().stream())
+            .distinct()
+            .filter(sched -> !mm.getCalendarForNow(sched).isEmpty())
+            .collect(Collectors.toSet());
+
+        List<Long> minionsInMaintMode = minions.stream()
+            .filter(minion -> minion.getMaintenanceScheduleOpt()
+            .map(sched -> schedulesInMaintMode.contains(sched)) // keep minions that have maintenance mode
+            .orElse(true)) // or that have no maintenance schedule whatsoever
+            .map(minion -> minion.getId())
+            .collect(toList());
+
+         List<MinionServer> logList = minions.stream()
+                 .filter(m -> !minionsInMaintMode.contains(m))
+                 .collect(Collectors.toList());
+         logSkippedMinions(logList);
+
+         return minionsInMaintMode;
```

Comment:
The body of the method looks incorrectly formatted. Please reformat in your IDE or use the following snippet.

```java
        MaintenanceManager mm = MaintenanceManager.instance();
        Set<MaintenanceSchedule> schedulesInMaintMode = minions.stream()
                .flatMap(minion -> minion.getMaintenanceScheduleOpt().stream())
                .distinct()
                .filter(sched -> !mm.getCalendarForNow(sched).isEmpty())
                .collect(Collectors.toSet());

        List<Long> minionsInMaintMode = minions.stream()
                .filter(minion -> minion.getMaintenanceScheduleOpt()
                        .map(sched -> schedulesInMaintMode.contains(sched)) // keep minions that have maintenance mode
                        .orElse(true)) // or that have no maintenance schedule whatsoever
                .map(minion -> minion.getId())
                .collect(toList());

        List<MinionServer> logList = minions.stream()
                .filter(m -> !minionsInMaintMode.contains(m))
                .collect(Collectors.toList());
        logSkippedMinions(logList);

        return minionsInMaintMode;
```